### PR TITLE
doc: mcumgr: Fix license

### LINKED ISSUE #78927: doc: mcumgr: Wrong license for mcumgr-client
**Describe the bug**
The license shown for the `mcumgr-client` tool is BSD, the actual license of the tool is apache

**To Reproduce**
Read

**Expected behavior**
Information to be correct

**Impact**
Probably none

### PR DESCRIPTION
**Pull request by jeppenodgaard**
mcumgr-client is Apache-2 and not BSD licensed.
See: https://github.com/vouch-opensource/mcumgr-client/blob/main/LICENSE

Fixes: https://github.com/zephyrproject-rtos/zephyr/issues/78927

### CODE REVIEW

**Review by nordicjm — DISMISSED** [review]

**Review by kartben — DISMISSED** [review]

File: doc/services/device_mgmt/mcumgr.rst [hunk 1]
```diff
@@ -55,7 +55,7 @@ project.
...
         | `AuTerm <https://github.com/thedjnK/AuTerm/>`_                                 | ✓       | ✓     | ✓   | ✕      | ✕        | ✓      | ✓         | ✓   | ✓  | ✓   | ✓    | ✓        | ✓  | ✓     | ✓    | ✓      | Application   | C++ (Qt)   | GPLv3   |
         +--------------------------------------------------------------------------------+---------+-------+-----+--------+----------+--------+-----------+-----+----+-----+------+----------+----+-------+------+--------+---------------+------------+---------+
-        | `mcumgr-client <https://github.com/vouch-opensource/mcumgr-client/>`_          | ✓       | ✓     | ✓   | ✕      | ✕        | ✓      | ✕         | ✕   | ✕  | ✓   | ✕    | ✕        | ✕  | ✕     | ✕    | ✕      | Application   | Rust       | BSD     |
+        | `mcumgr-client <https://github.com/vouch-opensource/mcumgr-client/>`_          | ✓       | ✓     | ✓   | ✕      | ✕        | ✓      | ✕         | ✕   | ✕  | ✓   | ✕    | ✕        | ✕  | ✕     | ✕    | ✕      | Application   | Rust       | Apache  |
```
Comment (henrikbrixandersen) [inline doc/services/device_mgmt/mcumgr.rst:58]:
```suggestion
        | `mcumgr-client <https://github.com/vouch-opensource/mcumgr-client/>`_          | ✓       | ✓     | ✓   | ✕      | ✕        | ✓      | ✕         | ✕   | ✕  | ✓   | ✕    | ✕        | ✕  | ✕     | ✕    | ✕      | Application   | Rust       | Apache-2.0  |
```

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]
Shouldn't the license say Apache-2.0?

**Comment by jeppenodgaard** [comment]
> Shouldn't the license say Apache-2.0?

I do not know. 

There's two other clients in the list that does not include the license version. I followed their example. The GPL license specifies the version, so the list is inconsistent. I guess that should be fixed.

File: doc/services/device_mgmt/mcumgr.rst (hunk 1 above)
Comment (jeppenodgaard) [reply doc/services/device_mgmt/mcumgr.rst]: Fixed

**Review by nordicjm — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

**Review by de-nordic — APPROVED** [review]

**Pull request merged**